pos_payment_terminal/static/tests/pos_payment_terminal_tests.js

```diff
@@ -0,0 +1,272 @@
+                    assert.expect(7);
+                    var self = this;
+
+                    self.env.pos.proxy.connection.rpc = async function (
+                        url,
+                        params,
+                        options
+                    ) {
+                        console.log("MOCK CALL TO", url, params, options);
+                        // We get back a transaction ID, that we can then poll for
+                        if (url === "/hw_proxy/payment_terminal_transaction_start") {
+                            return new Promise((resolve) => {
+                                resolve({transaction_id: "X1234"});
+                            });
+                        } else if (url === "/hw_proxy/status_json") {
+                            return promiseTimeout(
+                                options.timeout,
+                                new Promise(async (resolve) => {
+                                    await new Promise((r) => setTimeout(r, 800));
+                                    resolve({
+                                        test_driver: {
+                                            is_terminal: true,
+                                            transactions: [],
+                                        },
+                                    });
+                                })
+                            );
+                        }
+                        console.log("Unexpected call, ignoring");
+                    };
+                    assert.rejects(self.payment_terminal.send_payment_request());
+                }
+            );
+
+            QUnit.test(
+                "Sad flow 4 - transaction too slow to get started",
+                async function (assert) {
+                    assert.expect(7);
+                    var self = this;
+
+                    self.env.pos.proxy.connection.rpc = async function (
+                        url,
+                        params,
+                        options
+                    ) {
+                        console.log("MOCK CALL TO", url, params, options);
+                        // We get back a transaction ID, that we can then poll for
+                        if (url === "/hw_proxy/payment_terminal_transaction_start") {
+                            // Too long! > 10000 ms
+                            await new Promise((r) => setTimeout(r, 10000));
+                            return new Promise((resolve) => {
+                                resolve({transaction_id: "X1234"});
+                            });
+                        }
+                        console.log("Unexpected call, ignoring");
+                    };
+                    assert.rejects(self.payment_terminal.send_payment_request());
+                }
+            );
+
+            // Happy flow 2 : user cancels, then retries, and now it is OK, and we also dont have
+            // duplicate status polling active anymore because of our changes
```

Comment:
here still to make a test for the modifications that I did in the other PR for pos_payment_terminal.